scrape/target_test.go

```diff
@@ -517,6 +517,11 @@ func TestBucketLimitAppender(t *testing.T) {
 			limit:       3,
 			expectError: true,
 		},
+		{
+			h:           example,
+			limit:       4,
+			expectError: false,
+		},
```

Comment:
Should this test that the resolution was reduced?

Reply:
Yes, since in this test, original bucket count is 6, the smallest possible buckets count we can get is 4, we can't reduce more to get smaller buckets count any more. So I add this test case.

I will try to add more comment or change the test case's name to describe the meaning of the test.

Reply:
I'd add `expectBucketCount` to verify the bucket count after the limit check. This would make it more obvious the semantics that the limit check also alters the buckets.

Reply:
In addition to `expectBucketCount`, I also add `expectSchema`  to check if the schema is reduced as expected.